### ISSUE TITLE
Task: OpenWRT Packages Definitions

### ISSUE THREAD

**Issue by amsiotdeveloper** [OnionIoT/documentation#28]
---
name: Documentation Article Tasklist
about: The purpose of this template is to manage the workflow process for creating articles.
title: ''
labels: ''
assignees: ''

---
# **The Task**
### **Task Owner:**
Who owns this task and will approve the final deliverables? 

Lazar

###  1. How does this task relate to the value track(s)?
- All of the effort we put into improving our product lines should provide value to our customers.
- We need to be able to clearly articulate the value each improvement will provide in the following three value tracks:
        1. Development - having a clear understanding of OpenWRT and Opkg will help new developers grow their development skills with Omega products.
        2. Deployment
        3. Production
           
###  2. Create a short statement explaining the objectives and goals for this ticket.
- This article is targeting new users and is intended as an introduction to OpenWRT packages.
- introduction to OpenWRT and packages (Opkg)
- what is a package: Opkg
- provide a high level overview of OpwnWrt and Opkg and how the solution is utilized in the creation of packages.
- Explain how OpenWRT packages are helpful for your applications.
- Explain the significance of packages for easy development of customer applications.

https://docs.google.com/document/d/1ZIyQi8kKy0OXSdA9Eez5ANB2-X8rQgoluR8hZefMqbk/edit
(link will be updated to the Onion documentation working driver)
https://drive.google.com/drive/u/0/folders/1WVEGdUH6SeyA5MP5gxzzM9gh82SLoT6N

- use a diagram from the slide deck and make improvements as needed
https://docs.google.com/presentation/d/1tJqwJSvzzhEm91H4E9WzsjzKvjLeTyxV3QUcsryLzlw/edit#slide=id.p

### 3. **Reference Material:**
- Provide hyperlinks and description for helpful and related information.
- OpenWRT

###  1. What are the task owners expectations? 
- An introductory article that is targeting beginner level users and users new to Omega.

###  2. What does the task owner require for a successful completion of the ticket?
    i.e. sample code, drawings, documents etc.
  
- The article should be approximately 500 words
- It should be clear and easy to learn for first time users. 

# **Date Required By:**
###  1. When is the scope required to be completed?
- no specified date
- 
###  2. Is this a hard date?  Specify if the task is a dependency of another task or if it will impact a customer requirement.
- No dependencies 

# **The Workflow**
     
## **Backlog Column: List of Subtasks**
- [x] Subtask 1 (Workflow Reference 1): Idea dump from subject matter expert is completed.
- Create document of ideas for an outline of the new article. 
- Deliverable: google doc 
- [x] Subtask 2 (Workflow Reference 2): Technical Writer has createe the outline of article and high level ideas for for review.
- Revise and expand on the provided google doc
- Deliverable: google doc outline  
- [x] Subtask 3 (Workflow Reference 3): SME has reviewed and approved the ticket "Ready for Development" 

## **Ready for Development Column: List of Subtasks**
- [x] Subtask 1(Workflow Reference 4): PM has assigned the task to a technical writer.
- [x] Subtask 2(Workflow Reference 5): Technical writer has moved ticket to "In Progress" when work begins. 

## **In Progress Column: List of Subtasks**
- Technical writer is creating content.
- Technical writer will push the ticket to "In Review" when the articles is ready for review.
- An iterative review process is expected.

## **In Review Column: List of Subtasks**
- [x] Subtask 1(Workflow Reference 6): Initial Review has been **Approved** by PM.
- [x] Subtask 2(Workflow Reference 7): Review has been **Approved** by SME.
- SME moves ticket to "Approved for Production"

## **Approved for Production Column: List of Subtasks**
- [x] Subtask 1(Workflow Reference 8): Technical writer has sent Pull Request.
- [x] Subtask 2(Workflow Reference 9): SME has comitted the document to the repo.
- SME moves the ticket to "Completed" column.

**Comment by greenbreakfast** [comment]
This is a good placeholder issue. The scope may change depending on #25

**Comment by greenbreakfast** [comment]
@iammuze added comments to outline. please review

**Comment by greenbreakfast** [comment]
I've added another round of comments to the outline cc @iammuze

**Comment by iammuze** [comment]
Hi, @greenbreakfast I have attended the latest comments.

**Comment by greenbreakfast** [comment]
@iammuze thanks for the updates. I've added 2 more comments. Once you address them please go ahead with starting the article

**Comment by iammuze** [comment]
Hi @greenbreakfast , this doc is ready for review.

**Comment by iammuze** [comment]
@amsiotdeveloper Yeah, sure.

**Comment by amsiotdeveloper** [comment]
@iammuze I have reviewed the article and provided comments, feedback and suggestions within the document. 
I learned quite a bit and several things became more clear to me as I read so I think the content is good at this level.
However, please review item 2 in the scope of the ticket.  I feel the article is still lacking on these 2 items from the scope:

Explain how OpenWRT packages are helpful for your applications.
Explain the significance of packages for easy development of customer applications.

Please add more content or elaborate more on those 2 items.  

Because this is a slight change in the outline that @greenbreakfast already approved, then I think we should get some feedback from @greenbreakfast to support/modify my feedback accordingly on these 2 points. 

other than that, good work and I like the tone.

**Comment by iammuze** [comment]
@amsiotdeveloper, thank you for your thorough review. I'll address the points you've highlighted, I agree it's important to ensure alignment with the approved outline, so let's wait for input from @greenbreakfast before finalizing any adjustments.

**Comment by iammuze** [comment]
Hi @amsiotdeveloper I have resolved all the comments and have provided the necessary treatment. Can you please check again?

**Comment by greenbreakfast** [comment]
Weighing in on @amsiotdeveloper's note from [above](https://github.com/OnionIoT/documentation/issues/28#issuecomment-1917308632):
> However, please review item 2 in the scope of the ticket. I feel the article is still lacking on these 2 items from the scope:
> 
> Explain how OpenWRT packages are helpful for your applications. 
> Explain the significance of packages for easy development of customer applications.

The primary goal for this intro article is to briefly introduce the concepts discussed in this chapter and, importantly, how these concepts relate to each other.
We can have a second section that briefly talks about why packages are helpful and why it makes development easier.

For example, this article - https://www.linode.com/docs/guides/linux-package-management-overview/ - does a great job of **succinctly** introducing the concepts and the benefits.
As a user, I don't mind reading a couple of sentences about each concept to get an idea of what's going on.

**Comment by iammuze** [comment]
Thanks for your input, @greenbreakfast. I'll illustrate these points.

**Comment by greenbreakfast** [comment]
@iammuze, document reviewed, see comments

**Comment by iammuze** [comment]
Hi @greenbreakfast , comments have been addressed.

**Comment by greenbreakfast** [comment]
@iammuze, added another round of comments. Almost there!

**Comment by iammuze** [comment]
Understood Lazar

**Comment by iammuze** [comment]
hi @greenbreakfast , comments have been addressed. Could you please let me know the exact location in the `docs` directory where we should upload the `Getting Started Guide` document?

**Comment by greenbreakfast** [comment]
Hi @iammuze, please add this content to `docs/packages/intro.md` and `docs/packages/_definitions_packages.mdx`

**Comment by iammuze** [comment]
Hi, @amsiotdeveloper in 300dpi, both diagrams are covering the default width and height, as shown in these [screenshots](https://imgur.com/a/2Or6gEU). However, when we change the dpi, the diagrams are aligned to the left, and the .png files appear blurred. Could you please propose a solution?
